extensions/wikia/ImageReview/ImageListGetter.class.php

```diff
@@ -0,0 +1,312 @@
+<?php
+
+use Wikia\Logger\WikiaLogger;
+
+class ImageListGetter extends ImageReviewHelperBase {
+
+	/**
+	 * LIMIT_IMAGES_FROM_DB should be a little greater than LIMIT_IMAGES, so if
+	 * we fetch a few icons from DB, we can skip them
+	 */
+	const LIMIT_IMAGES = 20;
+	const LIMIT_IMAGES_FROM_DB = 128;
+	const IMAGE_REVIEW_THUMBNAIL_SIZE = 250;
+
+	private $userId = 0;
+	private $timestamp;
+	private $state;
+	private $order;
+	private $imageStateUpdater;
+	private $imageList = [];
+
+	function __construct( int $timestamp, int $state, int $order ) {
+		parent::__construct();
+		$this->userId = $this->wg->user->getId();
+		$this->timestamp = wfTimestamp( TS_DB, $timestamp );
+		$this->state = $state;
+		$this->order = $order;
+		$this->imageStateUpdater= new ImageStateUpdater();
+	}
+
+	/**
+	 * Get batch of images for Image Review Tool. Images are filtered based on
+	 * state (eg, unreviewed, rejected, questionable), order (last edited, priority),
+	 * and timestamp.
+	 *
+	 * When getting this images, we first check if the given timestamp matches
+	 * a previously reviewed batch of images. If so, we return those images. If not,
+	 * we then check to see if there are any outstanding reviews (reviews that this
+	 * reviewer started on, but didn't finished by making a decision on). If so, we
+	 * return those images. If not again, we'll finally create a new review and return
+	 * those images.
+	 */
+	public function getImageList() {
+		$this->fetchPreviousReviewFromTimestamp();
+
+		if ( !$this->fetchedEnoughImages() ) {
+			$this->fetchUnfinishedReview();
+		}
+
+		if ( !$this->fetchedEnoughImages() ) {
+			$this->fetchNewReview();
+		}
+
+		return $this->imageList;
+	}
+
+	private function fetchPreviousReviewFromTimestamp() {
+		$this->fetchAndPrepareImageResults( function() {
+			return ( new WikiaSQL() )
+				->SELECT_ALL()
+				->FROM( 'image_review' )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "image_review" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)